### PR TITLE
Use context manager for post request

### PR DESCRIPTION
**Pull request by kclowes**
@Uxio0 - I came up with something similar but without the manual management of the timeout. Does this do what you need it to? I can port to v7 if that's easier too.

### What was wrong?

Related to Issue #
Closes #

### Todo:

- [ ] Clean up commit history
- [ ] Add or update documentation related to these changes
- [ ] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: web3/_utils/request.py [hunk 1]
```diff
@@ -111,9 +110,16 @@ def get_response_from_post_request(
 def make_post_request(
     endpoint_uri: URI, data: Union[bytes, Dict[str, Any]], *args: Any, **kwargs: Any
 ) -> bytes:
-    response = get_response_from_post_request(endpoint_uri, data=data, *args, **kwargs)
-    response.raise_for_status()
-    return response.content
+    kwargs.setdefault("timeout", DEFAULT_TIMEOUT)
+    kwargs.setdefault("stream", True)
+    with get_response_from_post_request(
```
Comment (Uxio0) [inline web3/_utils/request.py:115]: Hi @kclowes . Looks good, but you are still missing the manual control of the `timeout`. In this case, even if you are streaming, if let's say, you set the timeout to `60 seconds`, and the RPC returns `1 byte` every `59 seconds`, the timeout will not trigger and this loop can be stuck until the RPC stops sending data

Comment (kclowes) [reply web3/_utils/request.py:115]: Gotcha. I'll close this is favor of yours then. On the one hand, it seems like this is desired behavior though? Could you decrease the value of your timeout?

Comment (kclowes) [reply web3/_utils/request.py:115]: Curious to get your thoughts here too @fselmo. Related to #3418

Comment (Uxio0) [reply web3/_utils/request.py:115]: Reducing the timeout is not helping if the server is slowly streaming a response (as Erigon does, for example)

**Pull request closed without merging**